test/NuGet.Core.Tests/NuGet.Packaging.Test/RelatedFileExtensionPropertyTests.cs

```diff
@@ -0,0 +1,197 @@
+            {
+                "lib/net50/system.dll"
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleAssemblyAssetsWithSamePrefix_GetNullProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.dll",
+                "lib/net50/system.exe",
+                "lib/net50/system.winmd"
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleAssemblyAssetsWithVariousPrefix_GetNullProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.dll",
+                "lib/net50/system.exe",
+                "lib/net50/system.Core.dll"
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleAssemblyAssetsWithVariousPrefixMixedCases_GetNullProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.dll";
+            string[] paths = new string[]
+            {
+                "lib/net50/system.dll",
+                "lib/net50/system.EXE",
+                "lib/net50/system.Core.DLL",
+                "lib/net50/system.NET.DLL",
+            };
+            collection.Load(paths);
+
+            // Act
+            string relatedProperty = collection.GetRelatedFileExtensionProperty(assembly, CreateAssetsFromPathList(paths));
+
+            // Assert
+            Assert.Equal(null, relatedProperty);
+        }
+
+        [Fact]
+        public void GetRelatedFileExtensionProperty_MultipleRelatedFilesWithSamePrefix_GetCorrectProperty()
+        {
+            // Arrange
+            var collection = new ContentItemCollection();
+            string assembly = "lib/net50/system.test.dll";
+            string[] paths = new string[]
```

Comment:
nit: You can probably use a Theory to make all these tests easier to read.

Reply:
Thanks! Updated to a Theory and also changed the case sensitive test cases.